### DISCUSSION TITLE
Best upgrade path

### DISCUSSION THREAD

**Discussion by tabuchid** [rmosolgo/graphql-ruby#4092] [Q&A]
Hello friends,
I'm in the process of upgrading a codebase that was on a very old version of graphql-ruby (Pre-class based apis). I've gotten the codebase safely to 1.12.24 which is where deprecation warnings are live but removal of the DSL style hasn't happened yet.
Do you have any advice on next steps in the upgrade process? I've been reading the documentation that is still on the 1.12.x branch but no longer live on the site to try and get my bearings. There are some upgrade tools that don't quite seem to work for our use case (Will try and get some specific examples later). 
Best to deal with warnings on 1.12 and continue to work towards 2.0.x or jump all the way up to 2.0 and work it all out at once?
If doing it piece by piece where are the best places to start? Seems like the the root objects aren't the place to start because they'll attempt to include older DSL style fields etc and fail.
Thanks!

**Comment by rmosolgo** [comment]
I did this migration at GitHub using the `Upgrader` in GraphQL-Ruby, along with some custom transforms, and IIRC it went in this order, migrating from `.define` to classes: 

1. Schema 
1. Scalars 
1. Unions
1. Input Objects 
1. Enums
1. interfaces 
1. Objects 
1. Root Types

Pretty much deploying between each step. Some of the steps (unions, interfaces, objects) were big enough that I released part-way through, for example, migrate 25 type definitions, then once those were working, redeploy. It's been a while, but I hope that helps your brainstorming!
